libyul/backends/wasm/WasmAST.h

```diff
@@ -50,7 +50,7 @@ using Expression = std::variant<
 	Block, If, Loop, Break, BreakIf, Return
 >;
 
-struct Literal { uint64_t value; };
+struct Literal { std::variant<uint32_t, uint64_t> value; };
```

Comment:
I think literals, variable declarations and parameters for functions (and their returns) should have a `type` taken from an enum. For literals, a variant could be enough, though.

Reply:
Yeah, that's what I was going to do for variables.  At first I added `type` to literals too but then realized that it's completely redundant and dropped it. For variables `type` is necessary because we don't have a value there.

We could also have `LiteralI64` and `LiteralI32` but this did not seem in line with how these structs are defined in Yul.